docs/v2_design.md

```diff
@@ -0,0 +1,179 @@
+auto b3 = 5 * si::activity[Bq] == 5 * si::frequency[Hz];
+```
+
+---
+
+## Arithmetics
+
+- Should the below compile?
+- If yes, what should be the final type?
+
+```cpp
+// Named and derived dimensions (same units)
+auto q1 = 10 * si::length[m] / (2 * si::time[s]) + 5 * si::speed[m / s];
+
+// Named and derived dimensions (different units)
+auto q2 = 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+
+// Different named dimensions
+auto q3 = 5 * si::activity[Bq] + 5 * si::frequency[Hz];
+
+// ???
+auto q3a = 5 * si::activity[Bq] + 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+```
+
+---
+
+## Comparisons: Speed
+
+```cpp
+// same dimension type & different unit
+static_assert(1000 * si::length[m] == 1 * si::length[km]);
+// different dimension type (derived unnamed vs named)
+static_assert(20 * si::length[m] / (2 * si::time[s]) == 10 * si::speed[m]); // ???
+
+static_assert(isq::length_dim == isq::length_dim);
+// derived unnamed vs named
+static_assert(equivalent(isq::length_dim / isq::time_dim, isq::speed_dim));
+static_assert(isq::length_dim / isq::time_dim != isq::speed_dim);           // ???
+// both sides return the same base dimension (at least for now)
+static_assert(equivalent(isq::speed_dim * isq::time_dim, isq::length_dim));
+static_assert(isq::speed_dim * isq::time_dim == isq::length_dim);           // ???
+// NOTE: "magnetomotive force" derived quantity is measured in ampere (which is a base unit)
+// Should the result of the above be just `isq::length_dim` or
+// `derived_dimension<isq::length_dim>` that can me converted to isq::length_dim?
+// In the latter case we would be able to make the above `false` and be able to
+// make "magnetomotive force" a dimension distinguishable from "electric current"
+
+static_assert(equivalent(si::metre, si::kilometre));
+static_assert(si::metre != si::kilometre);
+```
+
+---
+
+## Comparisons: Torque vs Energy
+
+```cpp
+// same named dimension & different but equivalent unit
+static_assert(10 * si::energy[N * m] == 10 * si::energy[J]);
+// unnamed derived dimension vs named dimension & equivalent units
+static_assert(5 * si::force[N] * (2 * si::length[m]) == 10 * si::energy[J]);   // ???
+
+// unnamed derived dimension vs named dimension
+static_assert(equivalent(isq::force_dim * isq::length_dim, isq::energy_dim));
+static_assert(isq::force_dim * isq::length_dim != isq::energy_dim);            // ???
+static_assert(equivalent(isq::force_dim * isq::length_dim, isq::torque_dim));
+static_assert(isq::force_dim * isq::length_dim != isq::torque_dim);            // ???
+// different (but dimensionally equivalent) dimensions
+static_assert(!equivalent(isq::energy_dim, isq::torque_dim));                  // ???
+static_assert(isq::energy_dim != isq::torque_dim);
+
+// unnamed derived unit vs named unit (note: inconsistent with dimensions)
+static_assert(equivalent(si::newton * si::metre, si::joule));
+static_assert(si::newton * si::metre == si::joule);                            // ???
```

Comment:
I am starting to think that it should not be equivalent. With this, it would be consistent with dimensions. `1/s`, `Hz`, and `Bq` are units used for different quantities defined in the ISO 80000.